logger/unilog.go

```diff
@@ -149,6 +153,83 @@ const (
 // Stats is Unilog's statsd client.
 var Stats *statsd.Client
 
+// tagPair is a simple pair of a tag t and the full metric name n
+type tagPair struct {
+	t string
+	n string
+}
+
+// independentTags stores a list of tags to individually emit metrics on.
+// Each metric name will be formatted exactly once and cached in metricsTable.
+type independentTags struct {
+	// The tags to build metric names from
+	// Format is foo:bar where foo is the tag name
+	Tags []string
+	// Lookup table for metricName -> slice of metricName.tagName
+	metricsTable map[string][]tagPair
+}
+
+func newIndependentTags(tags []string) *independentTags {
+	return &independentTags{Tags: tags, metricsTable: make(map[string][]tagPair)}
+}
+
+func setupIndependentTags() *independentTags {
+	return newIndependentTags(strings.Split(independenttags, ","))
+}
+
+func (it *independentTags) GetTags(key string) []tagPair {
```

Comment:
`metricName` might be clearer than `key` here

Reply:
👍